src/Shared/Microsoft.DotNet.Web.Authentication/DefaultAuthorizeActionModelConvention.cs

```diff
@@ -22,6 +23,13 @@ public DefaultAuthorizeActionModelConvention(string policyName)
 
         public void Apply(ActionModel action)
         {
+            // ASP.NET 3.1 broke IAllowAnonymousFilter, just find the attribute ourselves
+            // Otherwise it takes like 500 lines of code
+            if (action.ActionMethod?.GetCustomAttributes(true).Any(a => a is IAllowAnonymous || a is IAuthorizeData) ?? false)
```

Comment:
Unfortunately this won't work if we but `[AllowAnonymous]` on the controller level, but we just shouldn't do that.

Reply:
K.  There is some code later that reads the "Controller.Attributes" already, so I _think_ that should be fine already... it wasn't reading the filters from the controller, but the attributes themselves.